explorer/testdata/class/class_subtyping.carbon

```diff
@@ -0,0 +1,42 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
It seems like the main thing this is testing is non-virtual dispatch, rather than class subtyping more broadly, so maybe it should be named something like `non_virtual_dispatch.carbon`?